### PR TITLE
Enable native cosmetic filtering backend on production

### PR DESCRIPTION
**Pull request by antonok-edm**
Verified that this appears correctly on https://griffin.brave.com and in brave://version on Linux/Release when using the staging URL.

### CODE REVIEW

**Review by moritzhaller — APPROVED** [review]

**Review by kjozwiak — APPROVED** [review]
Approved & Merged after deliberating with @brave/uplift-approvers.

**Pull request merged**
